### PR TITLE
test: add unit tests for ContextLogic

### PR DESCRIPTION
**Pull request by sagar-rout**
#132 

Added unit tests for ContextLogic

### CODE REVIEW

File: src/test/java/com/crowdin/logic/ContextLogicTest.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+class ContextLogicTest {
+    private CrowdinConfig properties;
+    private VirtualFile root;
+    private CrowdinProjectCacheProvider.CrowdinProjectCache crowdinProjectCache;
+    private VirtualFile file;
+    private Map<String, FileInfo> filePaths;
+    private MockedStatic<FileUtil> fileUtilMockedStatic;
+    private MockedStatic<PlaceholderUtil> placeholderUtilMockedStatic;
+
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/logic/ContextLogicTest.java:32]:
```suggestion
```

File: src/test/java/com/crowdin/logic/ContextLogicTest.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        when(file.getPath()).thenReturn("/path/to/source/file");
+        when(root.getPath()).thenReturn("/path/to");
+        when(FileUtil.findRelativePath(root, file)).thenReturn("source/file");
+        when(FileUtil.sepAtStart("source/file")).thenReturn("/source/file");
+
+        RuntimeException exception = assertThrows(RuntimeException.class, () ->
+                ContextLogic.findSourceIdFromSourceFile(properties, filePaths, file, root));
+        assertTrue(exception.getMessage().contains("Couldn't find any crowdin representative for source file '/source/file'. Skipping"));
+    }
+
```
Comment (andrii-bodnar) [inline src/test/java/com/crowdin/logic/ContextLogicTest.java:79]:
```suggestion
```

**Review by andrii-bodnar — APPROVED** [review]
@sagar-rout thank you!

**Comment by andrii-bodnar** [comment]
@sagar-rout feel free to suggest new pull requests if you see more places that can be covered by unit tests.

**Comment by sagar-rout** [comment]
Hi @andrii-bodnar , It would be nice if you can create issues where you feel that it is more important for this project. As I am not that well versed with this thing. Thanks for approving and merging my PRs.

**Pull request merged**